src/sdk/pynni/nni/compression/torch/utils/shape_dependency.py

```diff
@@ -185,6 +185,49 @@ def dependency_sets(self):
             d_sets.append(tmp_set)
         return d_sets
 
+class InputChannelDependency(ChannelDependency):
+    def __init__(self, model, dummy_input=None, traced_model=None):
+        super(InputChannelDependency, self).__init__(model, dummy_input, traced_model)
+
+    def _get_following_convs(self, tensor):
+        queue = []
+        following_convs = []
+        queue.extend(self.graph.input_to_node[tensor])
+        while queue:
+            curnode = queue.pop(0)
+            if curnode.op_type == 'Conv2d' or curnode.op_type == 'Linear':
+                # find the first met conv
```

Comment:
`Linear` is not included into `following_convs`?

Reply:
Yes, in case the pruner may also want to prune the input channel of the linear layer.

Reply:
might be better to not use `folllowing_convs` then

Reply:
fixed~ thanks.